src/main/java/nextstep/subway/applicaion/LineService.java

```diff
@@ -3,109 +3,92 @@
+        Line line = getLineById(id);
+        line.update(request.getName(), request.getColor());
     }
 
-    public void deleteLine(Long id) {
+    public void deleteLineById(final Long id) {
         lineRepository.deleteById(id);
     }
 
-    public void addSection(Long lineId, SectionRequest sectionRequest) {
-        Station upStation = stationService.findById(sectionRequest.getUpStationId());
-        Station downStation = stationService.findById(sectionRequest.getDownStationId());
-        Line line = lineRepository.findById(lineId).orElseThrow(IllegalArgumentException::new);
+    public SectionResponse addSection(final SectionRequest request, final Long id) {
+        Line line = getLineById(id);
+        Station upStation = findStationById(request.getUpStationId());
+        Station downStation = findStationById(request.getDownStationId());
 
-        line.getSections().add(new Section(line, upStation, downStation, sectionRequest.getDistance()));
+        Section newSection = Section.of(line, upStation, downStation, request.getDistance());
+        line.addSection(newSection);
+        return SectionResponse.of(newSection);
     }
 
-    private LineResponse createLineResponse(Line line) {
-        return new LineResponse(
-                line.getId(),
-                line.getName(),
-                line.getColor(),
-                createStationResponses(line),
-                line.getCreatedDate(),
-                line.getModifiedDate()
-        );
+    public void deleteSection(final Long lineId, final Long downStationId) {
+        Line line = getLineById(lineId);
+        line.removeSection(downStationId);
     }
 
-    private List<StationResponse> createStationResponses(Line line) {
-        if (line.getSections().isEmpty()) {
-            return Collections.emptyList();
-        }
-
-        List<Station> stations = line.getSections().stream()
-                .map(Section::getDownStation)
-                .collect(Collectors.toList());
-
-        stations.add(0, line.getSections().get(0).getUpStation());
-
-        return stations.stream()
-                .map(it -> stationService.createStationResponse(it))
-                .collect(Collectors.toList());
+    private Line getLineById(final Long id) {
+        return lineRepository.findById(id)
+                .orElseThrow(NotFoundLineException::new);
     }
 
-    public void deleteSection(Long lineId, Long stationId) {
-        Line line = lineRepository.findById(lineId).orElseThrow(IllegalArgumentException::new);
-        Station station = stationService.findById(stationId);
-
-        if (!line.getSections().get(line.getSections().size() - 1).getDownStation().equals(station)) {
-            throw new IllegalArgumentException();
-        }
+    private boolean isDuplicate(final String lineName) {
+        Optional<Line> station = lineRepository.findByName(lineName);
+        return station.isPresent();
```

Comment:
아래와 같이 return해줘도 될 것 같아요! 
```suggestion
        return lineRepository.findByName(lineName)
                    .isPresent();
```